vm/src/types/program.rs

```diff
@@ -98,6 +95,12 @@ impl<'a> Arbitrary<'a> for SharedProgramData {
     }
 }
 
+#[derive(Clone, Default, Debug, PartialEq, Eq)]
+pub struct HintsCollection {
+    pub hints: Vec<HintParams>,
+    pub hints_ranges: Vec<HintRange>,
+}
```

Comment:
I'd suggest the following:
- The fields to be private;
- The struct to be pub(crate);
- Use a constructor taking a HashMap;
- Use a getter taking a range and returning a slice.

Reply:
Hi! Thanks for the suggestions! Just a short question.
The getter you are describing works fine for this case:
```
let hint_data = self 
     .program 
     .shared_program_data 
     .hints_ranges 
     .get(vm.run_context.pc.offset) 
     .and_then(|r| r.and_then(|(s, l)| hint_data.get(s..s + l.get()))) 
     .unwrap_or(&[]); 
```

But cases like this:
```
self.program
            .shared_program_data
            .hints_collection
            .hints
            .iter()
            .map(|hint| {
                hint_executor
                    .compile_hint(
                        &hint.code,
                        &hint.flow_tracking_data.ap_tracking,
                        &hint.flow_tracking_data.reference_ids,
                        references,
                    )
                    .map_err(|_| VirtualMachineError::CompileHintFail(hint.code.clone().into()))
            })
            .collect()
```

It seems a getter that returns all the hints is required. Let me know if I am wrong :)

Reply:
You can make a method that returns the iterator instead:
```rust
fn iter(&self) -> impl Iterator<Item = (pc, &[HintData])> {
    self.iter()
}
```

Reply:
You could also implement some `fn iter_hints(&self) -> impl Iterator<Item = &HintData>` if it makes this easier.

Reply:
Unfortunatelly I was made to implement both `iter_hints` and `iter`. The `iter` function is used only for testing in the helper `get_hints_as_map` thus it has `#[allow(dead_code)]`. Let me know your thoughts

Reply:
You can avoid the `#[allow(dead_code)]` by using a separate `impl` block that is gated by the `#[cfg(test)]` annotation.

Reply:
Added the new impl block! Thanks!